File: live-test/php/src/Probes.php
<?php
// Production half of the scope fixtures: Probe::target() has callers, and
// Probe has subtypes, both here and in the test tree — so production/test
// scope filters must return different sets.
namespace Demo;

class Probe
{
    public function target(): int
    {
        return 42;
    }

    // Caller inside the class itself: visible even at the narrowest scope.
    public function sameClassCaller(): int
    {
        return $this->target() + 1;
    }
}

// Production-scope caller outside the class.
function freeProdCaller(): int
{
    return (new Probe())->target();
}

// Production-scope subtype.
class ProbeProdChild extends Probe
{
}

File: live-test/python/src/probes.py
# Production half of the scope-discrimination fixtures; a twin in the test
# root adds the test-side caller and subclass. Scoped caller queries on
# target must discriminate: production keeps same_class_caller plus
# free_prod_caller, test keeps only the test-side caller, class scope only
# same_class_caller. Module scope does NOT narrow (the test caller still
# appears), and type-hierarchy scope is a no-op: both subclasses always show.
class Probe:
    def target(self):
        return 42

    def same_class_caller(self):
        return self.target() + 1


# Module-level production caller of target.
def free_prod_caller():
    return Probe().target()


# Production-side subclass.
class ProbeProdChild(Probe):
    pass
